### PR TITLE
[WIP] Complete jobs management

### PR DESCRIPTION
**Pull request by robkuijper**
## Description:

Adds jobs, fg and bg commands to vsh and supports '&' (background) processes.

**Related issue (if applicable):** fixes #<issue number goes here>

## Checklist:
  - [ ] The code change works
  - [ ] Passes all tests: `make test`
  - [ ] There is no commented out code in this PR.
  - [ ] `norminette srcs libft | grep -E "^Error" | wc -l` is not higher than master. If it is, run `norminette srcs libft | grep -E "^Error" -B 1` to see errors
  - [ ] I solemny swear my code is compliant with the [README][readme-file]

[readme-file]: https://github.com/OscarMulder/codam-42sh/blob/master/README.md

### CODE REVIEW

File: srcs/jobs/jobs_bg.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#include "vsh.h"
+#include <errno.h>
+#include <signal.h>
+
+void	jobs_bg_job(t_job *job, bool job_continued)
+{
+	jobs_print_job_info(job, JOB_OPT_L, g_data->jobs->joblist);
+	job->bg = true;
+	if (job_continued && kill(-job->pgid, SIGCONT) < 0)
+		ft_eprintf("kill (SIGCONT): %e\n", errno);
```
Comment (OscarMulder) [inline srcs/jobs/jobs_bg.c]: errno? Is this tmp?

File: srcs/jobs/jobs_fg.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+static int		job_stop(t_job *job)
+{
+	jobs_add_job(g_data, job);
+	if (g_state->shell_type == SHELL_INTERACT)
+		ft_printf("\r\x1b[0K^Z\n");
+	job->bg = true;
+	jobs_print_job_info(job, JOB_OPT_L, g_data->jobs->joblist);
+	return (146);
```
Comment (OscarMulder) [inline srcs/jobs/jobs_fg.c]: 🤔

File: srcs/jobs/jobs_fg.c [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+int				jobs_fg_job(t_job *job, bool job_continued)
+{
+	job->bg = false;
+	if (g_state->shell_type == SHELL_INTERACT)
+		tcsetpgrp(STDIN_FILENO, job->pgid);
+	tcsetattr(STDIN_FILENO, TCSADRAIN, g_data->term->old_termios_p);
+	if (job_continued == true)
+	{
+		if (kill(-job->pgid, SIGCONT) < 0)
+			return (ft_eprintf("Could not SIGCONT job %i\n", job->pgid));
```
Comment (OscarMulder) [inline srcs/jobs/jobs_fg.c]: Define

File: srcs/jobs/jobs_fg.c [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+/*   By: rkuijper <rkuijper@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2019/10/18 17:12:11 by rkuijper       #+#    #+#                */
+/*   Updated: 2019/10/30 13:35:11 by rkuijper      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "vsh.h"
+#include <term.h>
+#include <errno.h>
```
Comment (OscarMulder) [inline srcs/jobs/jobs_fg.c]: Do you use this?

File: srcs/jobs/jobs_launch_proc.c [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		close(fds[2]);
+	}
+}
+
+static void	execute_proc(t_proc *proc)
+{
+	if (proc->is_builtin == false)
+	{
+		execve(proc->binary, proc->argv, proc->env);
+		ft_eprintf("Error executing %s\n", proc->binary);
```
Comment (OscarMulder) [inline srcs/jobs/jobs_launch_proc.c]: Define

File: srcs/jobs/jobs_mark.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+				if (proc->pid == pid)
+				{
+					proc->exit_status = status;
+					if (WIFSTOPPED(status))
+						proc->state = PROC_STOPPED;
+					else
+					{
+						proc->state = PROC_COMPLETED;
+						if (WIFSIGNALED(status))
+							ft_eprintf("%d: Terminated by signal %d\n", proc->pid,
```
Comment (OscarMulder) [inline srcs/jobs/jobs_mark.c]: Define

File: srcs/jobs/jobs_wait.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+/*                                                     +:+                    */
+/*   By: rkuijper <rkuijper@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2019/10/21 11:22:46 by rkuijper       #+#    #+#                */
+/*   Updated: 2019/10/28 20:25:58 by jbrinksm      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "vsh.h"
+#include <errno.h>
```
Comment (OscarMulder) [inline srcs/jobs/jobs_wait.c]: Errnooooooo

File: srcs/jobs/jobs_fg.c (hunk 2 above)
Comment (robkuijper) [reply srcs/jobs/jobs_fg.c]: Yes, this will get a comment in the near future.

File: srcs/jobs/jobs_bg.c (hunk 1 above)
Comment (robkuijper) [reply srcs/jobs/jobs_bg.c]: Yes, until I find a better solution.

**Pull request merged**